jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1554,12 +1554,10 @@ code_blockt java_bytecode_convert_methodt::convert_instructions(
       const bool is_assertions_disabled_field=
         field_name.find("$assertionsDisabled")!=std::string::npos;
 
+      const irep_idt field_id(
+        get_static_field(arg0.get_string(ID_class), field_name));
       const symbol_exprt symbol_expr(
-        get_static_field(arg0.get_string(ID_class), field_name), arg0.type());
-
-      INVARIANT(
```

Comment:
:question: This looks like useful context for this symbol not existing that goes away by folding it into `lookup_ref` - do you disagree?

Reply:
Seems pointless to have an invariant that can never fail, and I think a symbol table lookup returning a reference is widely recognised as find-or-die?

Reply:
My point is `"getstatic symbol should have been created before method conversion"` is more useful than simply a symbol not existing. Like without that message you might (wrongly) conclude that this conversion should actually be adding to the symbol table

I'm suggesting the invariant goes about the call to `lookup_ref` in case that wasn't clear. Like,

Reply:
Added a clarifying comment